assignment-2/src/presenter/loginPresenter.js

```diff
@@ -0,0 +1,33 @@
+import user from "../model/Users";
+
+class QuestionsListPresenter {
+    onLogin(){
+        let newUser = user.state.newUser;
+        let users = user.state.users;
+
+        for (let i = 0; i < users.length; i++) {
+            if (newUser.username === users[i].username) {
+                if (newUser.password === users[i].password) {
+                    user.state.currentUser.username = newUser.username;
```

Comment:
Also, you are manipulating the state directly in the presenter + without creating a copy of the state object. As discussed at the lab, it is a very good idea to clone the state as in our examples instead of directly assigning to properties of the state.